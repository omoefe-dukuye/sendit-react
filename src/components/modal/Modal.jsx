import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

class Modal extends Component {
  componentDidMount () {
    document.addEventListener('click', this.handleToggle);
  }

  componentWillUnmount () {
    document.removeEventListener('click', this.handleToggle);
  }

  handleToggle = ({ target }) => {
    if (this.modal.contains(target)) return;

    this.props.hideModal();
  };

  render() {
    const { content } = this.props;

    return (
      <div className="modal__background">
        <div className="modal__modal" ref={node => this.modal = node}>
          {content}
        </div>
      </div>
    );
  }
}

Modal.propTypes = {
  content: PropTypes.object,
  hideModal: PropTypes.func
};

export default Modal;
