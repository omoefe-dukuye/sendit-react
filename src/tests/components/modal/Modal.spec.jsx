import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../../../components/modal';

test('Should render static component', () => {
  const wrapper = shallow(<Modal />);
  expect(wrapper).toMatchSnapshot();
});
