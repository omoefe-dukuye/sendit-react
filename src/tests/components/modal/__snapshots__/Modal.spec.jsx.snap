// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render static component 1`] = `
<div
  className="modal__background"
>
  <div
    className="modal__modal"
  />
</div>
`;
